pkg/kubefed2/unjoin.go

```diff
@@ -163,76 +164,25 @@ func UnjoinCluster(hostConfig, clusterConfig *rest.Config, federationNamespace,
 		return err
 	}
 
-	glog.V(2).Infof("Performing preflight checks for unjoin cluster: %s.", unjoiningClusterName)
-	err = performPreflightChecksForUnjoin(clusterClientset, unjoiningClusterName, hostClusterContext, federationNamespace)
-	if err != nil {
-		return err
-	}
-
 	if removeFromRegistry {
-		err = removeFromClusterRegistry(hostConfig, clusterNamespace, clusterConfig.Host, unjoiningClusterName, dryRun)
-		if err != nil {
-			return err
-		}
-	}
-
-	glog.V(2).Infof("Deleting federated cluster resource from namespace: %s for unjoin cluster: %s",
-		federationNamespace, unjoiningClusterName)
-
-	_, err = deleteFederatedClusterAndSecret(hostClientset, fedClientset, federationNamespace,
-		unjoiningClusterName, dryRun)
-	if err != nil {
-		glog.V(2).Infof("Failed to delete federated cluster resource from namespace: %s for unjoin cluster: %s due to: %v",
-			federationNamespace, unjoiningClusterName, err)
-		return err
+		removeFromClusterRegistry(hostConfig, clusterNamespace, clusterConfig.Host, unjoiningClusterName, dryRun)
 	}
 
-	glog.V(2).Infof("Deleted federated cluster resource from namespace: %s for unjoin cluster: %s",
-		federationNamespace, unjoiningClusterName)
-
-	err = deleteRBACResources(hostClientset, clusterClientset,
-		federationNamespace, unjoiningClusterName, hostClusterContext, dryRun)
-	if err != nil {
-		glog.V(2).Infof("Could not delete cluster RBAC resources: %v", err)
-		return err
-	}
+	isSucceed := deleteRBACResources(clusterClientset, federationNamespace, unjoiningClusterName, hostClusterContext, dryRun)
```

Comment:
`s/isSucceed/deletionSucceeded/`

`isSucceed` is not grammatically correct in English.